third_party/amd/lib/TritonAMDGPUToLLVM/TargetInfo.h

```diff
@@ -34,6 +34,8 @@ class TargetInfo : public mlir::triton::TargetInfoBase {
                       int swizzleByteSize) const override;
   void storeMatrixShared(RewriterBase &rewriter, Location loc, Value ptr,
                          Value val) const override;
+  virtual Value loadMatrixShared(RewriterBase &rewriter, Location loc,
```

Comment:
is there any specific reason to add it to TargetInfo if it is only used by nvidia backend? I believe stmatrix was there because the code for stmatrix wasn't fully decoupled but if the implementation of ldmatrix is fully in nvidia specific files I'm not sure there is an advantage having it here.

Reply:
> I believe stmatrix was there because the code for stmatrix wasn't fully decoupled 

The hope is that we may use `ldmatrix` similar to `stmatrix` in ConvertLayoutToLLVMPass.cpp, which relies on TargetInfo

Reply:
I agree it's not used at this moment

Reply:
> The hope is that we may use ldmatrix similar to stmatrix in ConvertLayoutToLLVMPass.cpp, which relies on TargetInfo

It still feels like adding nvidia specific code in the generic pattern. We could have a separate pattern woth higher benefit for nvidia instead in order to separate the two

Reply:
OK, let me take it out